### ISSUE TITLE
Sort packages in patches section following sort-packages config

### ISSUE THREAD

**Issue by JPustkuchen** [cweagans/composer-patches#341]
### Problem:
If you have to handle a lot of patches in large projects or even have to update / diff them on a regular basis, a logical order of the packages patched helps a lot. Currently, the user has to sort the patches by package name himself.

### Proposed solution:
It would be very handy to have an option to automatically update the patches order by package name like

```
"config": {
    "sort-packages": true,
} 
```

for packages.

Great work, your plugin is one of the very most important for composer ever!

**Comment by Pasqualle** [comment]
Some notes:
1. "sort-packages" is an existing composer config. I guess patches would not require additional config parameter. But that would mean a backward compatibility break, therefore a new major version might be required.
2. Patch files should never be sorted, only the packages where patches are applied. Sorting patch files can break the patching if multiple patches are defined for the same package.
3. patches can be in external composer.patches.json file, which might complicate the solution.

**Comment by Pasqualle** [comment]
After thinking about it more, sorting the list of packages where patches will be applied, can break the patching in case patches from dependencies are allowed.

**Comment by cweagans** [comment]
Probably not going to do this. The order that patches are applied to a dependency is not something that can just be changed in most cases.

**Comment by leymannx** [comment]
@cweagans @Pasqualle – This issue not about the order of patches but about the order of packages in the patches section. Same as the order of packages in the require section. In a project with lots of patches having to adjust this section, alphabetical order helps a lot.

**Comment by Pasqualle** [comment]
Yes, I agree, packages should be sorted automatically, same as composer does it.
Sorry if my comment was misleading. I just tried to explain the problem with "sort-patches" word in the issue title.

Please re-open as the rejection reason is invalid. Packages can (and should) be sorted.

**Comment by leymannx** [comment]
@JPustkuchen – Can you maybe rephrase the title? Something like "Sort packages in patches section following sort-packages config" or similar?

**Comment by cweagans** [comment]
I don't have time to work on this, but PRs are welcome!
